libc/LibC.h

```diff
@@ -114,10 +109,10 @@ Simplifies creation of the memory context in constructors and includes error han
     /* Free context and croak on error */                                                                                          \
     if (MEM_CONTEXT_XS_croak)                                                                                                      \
     {                                                                                                                              \
-        memContextFree(MEM_CONTEXT_XS_memContext);                                                                                 \
-                                                                                                                                   \
         ERROR_XS()                                                                                                                 \
     }                                                                                                                              \
+    else                                                                                                                           \
+        memContextKeep();                                                                                                          \
```

Comment:
In the C code, MEM_CONTEXT_NEW_BEGIN does memContextNew() and memContextPush() and then MEM_CONTEXT_NEW_END does memContextPop() and memContextKeep().  So here, in MEM_CONTEXT_XS_NEW_BEGIN there is also a New and a Push but the MEM_CONTEXT_XS_NEW_END does not do a Pop() only a Keep() - why?

Reply:
Same reason as above.